Assets/MRTK/SDK/Experimental/Drawstring/Scripts/ElasticHandleLogic.cs

```diff
@@ -0,0 +1,123 @@
+    /// distance between the handles.
+    /// </summary>
+    internal class ElasticHandleLogic
+    {
+        private Vector3 leftInitialPosition = Vector3.zero;
+        private Vector3 rightInitialPosition = Vector3.zero;
+
+        private LinearElasticSystem elasticSystem = null;
+
+        private bool isSetup;
+
+        /// <summary>
+        /// Initialize system with source info from controllers/hands.
+        /// Left and right handles must both be specified, but this does not
+        /// preclude single-hand interaction when Update() is called.
+        /// </summary>
+        /// <param name="leftHandleStart">World position of "left" handle point.</param>
+        /// <param name="rightHandleStart">World position of "right" handle point.</param>
+        /// <param name="extentInfo">Properties of the linear 1-D elastic extent to be manipulated</param>
+        /// <param name="elasticProperties">Properties of the elastic material/spring.</param>
+        /// <param name="leftHandleVelocity">Optional, initial velocity in 1-dimensional stretch space</param>
+        /// <param name="rightHandleVelocity">Optional, initial velocity in 1-dimensional stretch space</param>
+        public virtual void Setup(Vector3 leftHandleStart, Vector3 rightHandleStart,
+                                    ElasticExtentProperties<float> extentInfo, ElasticProperties elasticProperties,
+                                    float leftHandleVelocity = 0.0f, float rightHandleVelocity = 0.0f)
+        {
+            isSetup = true;
+            leftInitialPosition = leftHandleStart;
+            rightInitialPosition = rightHandleStart;
+            elasticSystem = new LinearElasticSystem((leftHandleStart - rightHandleStart).magnitude, 0.0f, extentInfo, elasticProperties);
+        }
+
+        /// <summary>
+        /// Update the internal state of the damped harmonic oscillator, given the left and right pointer positions.
+        /// Note, both left and right input positions are nullable; set the pointer position to null
+        /// if only one of the handles is being interacted with. Returns the calculated distance between the
+        /// handles.
+        /// 
+        /// </summary>
+        /// <param name="leftPointer">World position of the pointer manipulating the left handle.</param>
+        /// <param name="rightPointer">World position of the pointer manipulating the right handle.</param>
+        /// <param name="deltaTime">Amount of time to simulate this step.</param>
+        /// <param name="normalizedAxis">World axis of the two-handle system.</param>
+        public virtual float Update(Vector3? leftPointer, Vector3? rightPointer, float deltaTime, Vector3? normalizedAxis = null)
+        {
+            // If we have not been Setup() yet
+            if (!isSetup) { return 0.1f; }
+
+            var handleAxis = normalizedAxis ?? (rightInitialPosition - leftInitialPosition).normalized;
```

Comment:
If I'm reading this correctly as being static-to-setup, is there a reason why we'd need to recompute the normalzed right-left thing each time?

Reply:
If only because normalized involves some sqrting and other calculations.